tests/api/test_criteria.py

```diff
@@ -192,8 +194,17 @@ def test_post_criterion_category_doesnt_exist(self, mock_auth):
         data = {'category_id': category_id}
 
         response = self.client.post('/criteria', json=data, headers=auth_headers())
-        self.assertEqual(response.status_code, 404)
-        mock_auth.assert_called_once()
+        self.assertEqual(response.status_code, 400)
+
+        json_response = json.loads(response.data)
+        self.assertEqual(json_response['description'], strings.category_not_found)
+
+    @patch('auth.is_token_valid', return_value=True)
+    def test_post_criterion_no_category(self, mock_auth):
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', category=SAWarning)
+            response = self.client.post('/criteria', json={}, headers=auth_headers())
```

Comment:
This is needed because `update_or_create_criterion` will try to create a criterion with `categoryId=None`, which causes [this warning](https://github.com/zzzeek/sqlalchemy/blob/3b8a14153da9e7b6694571fa10f6d30c4012ee82/lib/sqlalchemy/orm/loading.py#L424-L429).

It feels a bit hacky - let me know if you know a better way to do this!

Reply:
I do not know a better way to do this and it also makes sense to me 🤷‍♀️ I'll do some thinking about it but I don't think it's a blocker!